### PR TITLE
feat: delete functionality for product

### PR DESCRIPTION
**Pull request by DPDeepak**
Ticket : https://github.com/nodexpertsdev/RnD/issues/121

### CODE REVIEW

File: expressEngine/product/service/product.js [hunk 1]
```diff
@@ -25,6 +25,20 @@ class Service {
...
+      isDiscontinued: true,
+    }, { productId: id });
+    const { n, nModified } = product;
+    if (nModified) {
+      return { message: success.productDeleted };
+    } if (n) {
+      return { message: success.productAlreadyDeleted };
+    }
+    return { message: success.productNotFound };
+  }
```
Comment (chiragSuccessive) [inline expressEngine/product/service/product.js]: db errors are not handled.

File: expressEngine/product/service/product.js [hunk 2]
```diff
@@ -25,6 +25,20 @@ class Service {
...
+    const product = await DBService.updateOne(Product, {
+      isDiscontinued: true,
+    }, { productId: id });
+    const { n, nModified } = product;
+    if (nModified) {
+      return { message: success.productDeleted };
+    } if (n) {
+      return { message: success.productAlreadyDeleted };
+    }
+    return { message: success.productNotFound };
```
Comment (chiragSuccessive) [inline expressEngine/product/service/product.js]: @DPDeepak productNotFound should be returned as an error.

File: expressEngine/cms/product/success.js [hunk 3]
```diff
@@ -1,4 +1,7 @@
 export default {
   userRegistered: 'You have registered successfully',
   productRegistered: 'Product registered successfully',
+  productDeleted: 'Successfully deleted Product',
+  productAlreadyDeleted: 'Product already deleted',
+  productNotFound: 'ProductNotFound',
```
Comment (chiragSuccessive) [inline expressEngine/cms/product/success.js]: @DPDeepak replace 'ProductNotFound' with 'Product not found'.

File: expressEngine/product/service/product.js (hunk 1 above)
Comment (DPDeepak) [reply expressEngine/product/service/product.js]: done

File: expressEngine/product/service/product.js (hunk 2 above)
Comment (DPDeepak) [reply expressEngine/product/service/product.js]: done

File: expressEngine/cms/product/success.js (hunk 3 above)
Comment (DPDeepak) [reply expressEngine/cms/product/success.js]: done

**Comment by gkgbunny** [comment]
Please tag yourself as assignee and remove conflicts

File: expressEngine/model/collection/product.js [hunk 4]
```diff
@@ -30,15 +30,18 @@ const productsSchema = new Schema({
     required: false,
   },
 
-}, { collection: 'product', timestamps: true });
+}, {
+    collection: 'product',
+    timestamp: true,
```
Comment (aman-successive) [inline expressEngine/model/collection/product.js]: change timestamp to timestamps

File: expressEngine/model/collection/product.js [hunk 5]
```diff
@@ -30,15 +30,18 @@ const productsSchema = new Schema({
...
+  });
 
 productsSchema.pre('save', function preSave() {
   const product = this;
   product.id = product._id.toString();
 });
 
+const Product = model('Products', productsSchema);
```
Comment (aman-successive) [inline expressEngine/model/collection/product.js]: use mongoose.model, using only model shoes not defined error

File: expressEngine/model/collection/product.js (hunk 4 above)
Comment (DPDeepak) [reply expressEngine/model/collection/product.js]: done

File: expressEngine/model/collection/product.js (hunk 5 above)
Comment (DPDeepak) [reply expressEngine/model/collection/product.js]: i have import model from mongoose.

Comment (DPDeepak) [reply expressEngine/model/collection/product.js]: done

**Pull request merged**